src/publish.rs

```diff
@@ -72,7 +73,8 @@ where
     ) - bitcoin_wallet.bitcoin_fees();
 
     let rate = spread.apply(mid_market_rate);
-    let buy_amount = sell_amount.is_worth_in(rate);
+    // TODO: Return result
+    let buy_amount = sell_amount.is_worth_in(rate).unwrap();
```

Comment:
Would it make sense to calculate the `buy_amount` first and then apply the spread to it?

I am asking because I remember coding on the kraken interface with @da-kami and I think the actual data we get returned is _not_ a rate but a quantity. Meaning there would be quite some back and forth where we:

- ask for a the rate from kraken, we get `1BTC = 9543 DAI`
- convert that to a rate
- apply a spread to the rate
- convert the rate back to an amount

From that perspective, it seems like we could spare ourselves quite some arithmetic if we just do:

- ask for a the rate from kraken, we get `1BTC = 9543 DAI`
- apply a spread to the amount

Reply:
According to https://github.com/coblox/nectar/pull/14/files we get a float point rate from Kraken.

I applied the spread on the rate to limit the float point arithmetic and isolate it to that this unique operation: apply spread on rate.

If I were to apply the rate and then later the spread I'd end up with more float point arithmetic.

Reply:
> According to [#14 (files)](https://github.com/coblox/nectar/pull/14/files) we get a float point rate from Kraken.

Not quite, Daniel calculates it from responses like these: https://github.com/coblox/nectar/pull/14/files#diff-c4a4631dcbd8719f32d727ef48b9ce7fR117-R144

Reply:
> If I were to apply the rate to get the buy amount and then apply the spread I'd end up with 2 operations doing float point arithmetic.

Not if the input is already the buy-amount and you just need to apply the spread.

That is also just a multiplication then we doesn't need to involve floating point arithmetic.

Reply:
> > According to [#14 (files)](https://github.com/coblox/nectar/pull/14/files) we get a float point rate from Kraken.
> 
> Not quite, Daniel calculates it from responses like these: [#14 (files)](https://github.com/coblox/nectar/pull/14/files#diff-c4a4631dcbd8719f32d727ef48b9ce7fR117-R144)

The link does not seem to work. I see floating point values in there:
```
    fn ohlc_with_vwap() -> Ohlc {
        // These are actual values from Kraken
        Ohlc {
            timestamp: Utc::now(),
            high: 9825.1,
            low: 9791.0,
            vwap: 9806.7,
            trading_pair: TradingPair::BtcDai,
        }
    }
```

It's a rate but it is a price.

Reply:
> That is also just a multiplication then we doesn't need to involve floating point arithmetic.

Are you saying that multiplying floating points does not involve floating point arithmetic?

> ask for a the rate from kraken, we get 1BTC = 9543 DAI
> apply a spread to the amount

How do apply a spread to this amount without doing float point arithmetic?

> That is also just a multiplication then we doesn't need to involve floating point arithmetic.

It's a multiplication that involves decimal, so yeah, it does involves floating point arithmetic.

I think you should make your point across via code.